### PR TITLE
Glue datasource support

### PR DESCRIPTION
**Pull request by vamsi-amazon**
### Description
* Adding Basic Requirements for Glue datasource type.
* Will change properties again during end to end integration.
* Since Glue queries are dependent on jobs apis, we are not creating a separate module. For now keeping glue classes inside datasources module itself.
* Documentation link: https://github.com/vamsi-amazon/sql/blob/datasource-glue-support/docs/user/ppl/admin/connectors/s3glue_connector.rst [Will update this document on the go according to the changes made in subsequent PRs]

Testing:
```
curl --location 'http://localhost:9200/_plugins/_query/_datasources' \
--header 'Content-Type: application/json' \
--data '{
    "name" : "my_glue",
    "connector" : "glue",
    "properties" : {
           "glue.auth.type": "iam_role",
    "glue.auth.account_id": "access_key",
    "glue.auth.role_arn": "role_arn",
    "glue.auth.region": "region",    
    "emr.datasource.flint.uri": "http://localhost:9200",
    "emr.datasource.flint.auth" :"false",
    "emr.datasource.flint.region": "us-west-2"
    }
}'
```

Response:

```
Created DataSource with name my_glue
```

 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: common/src/main/java/org/opensearch/sql/common/setting/Settings.java [hunk 1]
```diff
@@ -35,7 +35,6 @@ public enum Key {
 
     METRICS_ROLLING_WINDOW("plugins.query.metrics.rolling_window"),
     METRICS_ROLLING_INTERVAL("plugins.query.metrics.rolling_interval"),
-
```
Comment (penghuo) [inline common/src/main/java/org/opensearch/sql/common/setting/Settings.java:38]: no change. revert it.

File: core/src/main/java/org/opensearch/sql/datasource/model/DataSourceType.java [hunk 2]
```diff
@@ -8,7 +8,9 @@
 public enum DataSourceType {
   PROMETHEUS("prometheus"),
   OPENSEARCH("opensearch"),
-  SPARK("spark");
+  SPARK("spark"),
+  GLUE("glue");
```
Comment (penghuo) [inline core/src/main/java/org/opensearch/sql/datasource/model/DataSourceType.java:12]: what is we name it as s3?

File: datasources/src/main/java/org/opensearch/sql/datasources/glue/GlueDatasourceFactory.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+  private final Settings pluginSettings;
+
+  // Glue configuration properties
+  public static final String GLUE_AUTH_TYPE = "glue.auth.type";
+  public static final String GLUE_ACCOUNT_ID = "glue.auth.account_id";
+  public static final String GLUE_ROLE_ARN = "glue.auth.role_arn";
+  public static final String GLUE_REGION = "glue.auth.region";
+
+  public static final String FLINT_URI = "emr.datasource.flint.uri";
```
Comment (penghuo) [inline datasources/src/main/java/org/opensearch/sql/datasources/glue/GlueDatasourceFactory.java:26]: why emr as prefix?

File: datasources/src/main/java/org/opensearch/sql/datasources/glue/GlueDatasourceFactory.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+import org.opensearch.sql.storage.DataSourceFactory;
+
+@RequiredArgsConstructor
+public class GlueDatasourceFactory implements DataSourceFactory {
+
+  private final Settings pluginSettings;
+
+  // Glue configuration properties
+  public static final String GLUE_AUTH_TYPE = "glue.auth.type";
+  public static final String GLUE_ACCOUNT_ID = "glue.auth.account_id";
```
Comment (penghuo) [inline datasources/src/main/java/org/opensearch/sql/datasources/glue/GlueDatasourceFactory.java:22]: glue.account_id?
not sure, can we support cross-account access?

File: datasources/src/main/java/org/opensearch/sql/datasources/glue/GlueDatasourceFactory.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  public static final String GLUE_ROLE_ARN = "glue.auth.role_arn";
+  public static final String GLUE_REGION = "glue.auth.region";
```
Comment (penghuo) [inline datasources/src/main/java/org/opensearch/sql/datasources/glue/GlueDatasourceFactory.java:24]: glue.region?

File: datasources/src/main/java/org/opensearch/sql/datasources/glue/GlueDatasourceFactory.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+import lombok.RequiredArgsConstructor;
+import org.opensearch.sql.common.setting.Settings;
+import org.opensearch.sql.datasource.model.DataSource;
+import org.opensearch.sql.datasource.model.DataSourceMetadata;
+import org.opensearch.sql.datasource.model.DataSourceType;
+import org.opensearch.sql.datasources.utils.DatasourceValidationUtils;
+import org.opensearch.sql.storage.DataSourceFactory;
+
+@RequiredArgsConstructor
+public class GlueDatasourceFactory implements DataSourceFactory {
```
Comment (dai-chen) [inline datasources/src/main/java/org/opensearch/sql/datasources/glue/GlueDatasourceFactory.java:16]: np: GlueData[S]ource

File: datasources/src/main/java/org/opensearch/sql/datasources/glue/GlueDatasourceFactory.java [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+  private final Settings pluginSettings;
+
+  // Glue configuration properties
+  public static final String GLUE_AUTH_TYPE = "glue.auth.type";
+  public static final String GLUE_ACCOUNT_ID = "glue.auth.account_id";
+  public static final String GLUE_ROLE_ARN = "glue.auth.role_arn";
+  public static final String GLUE_REGION = "glue.auth.region";
+
+  public static final String FLINT_URI = "emr.datasource.flint.uri";
+  public static final String FLINT_AUTH = "emr.datasource.flint.auth";
```
Comment (dai-chen) [inline datasources/src/main/java/org/opensearch/sql/datasources/glue/GlueDatasourceFactory.java:27]: could you add the meaning of all these configs to doc?

File: core/src/main/java/org/opensearch/sql/datasource/model/DataSourceType.java (hunk 2 above)
Comment (vamsi-amazon) [reply core/src/main/java/org/opensearch/sql/datasource/model/DataSourceType.java:12]: changed to s3Glue

File: datasources/src/main/java/org/opensearch/sql/datasources/glue/GlueDatasourceFactory.java (hunk 7 above)
Comment (vamsi-amazon) [reply datasources/src/main/java/org/opensearch/sql/datasources/glue/GlueDatasourceFactory.java:27]: will add a new page for glue datasource connector page.

File: datasources/src/main/java/org/opensearch/sql/datasources/glue/GlueDatasourceFactory.java (hunk 3 above)
Comment (vamsi-amazon) [reply datasources/src/main/java/org/opensearch/sql/datasources/glue/GlueDatasourceFactory.java]: Changed to glue.remotestore.flint.uri

**Review by penghuo — APPROVED** [review]

**Pull request merged**